engine/src/main/java/org/hibernate/validator/internal/metadata/core/MetaConstraint.java

```diff
@@ -41,13 +45,28 @@
 	private final ConstraintLocation location;
 
 	/**
+	 * The {@link ValueExtractor} used to extract the value for validation.
+	 */
+	private final ValueExtractorDescriptor valueExtractorDescriptor;
+
+	/**
+	 * The type of the validated element.
+	 */
+	private final Type typeOfValidatedElement;
```

Comment:
The `typeOfValidatedElement` is the type of the validation object. It might not be the type used for validator resolution as it might be the wrapped type.

Note that it's exactly the same behavior as before. The code has been moved but is exactly identical to the existing logic.